test/message_server/component.adb

```diff
@@ -0,0 +1,71 @@
+
+with Componolit.Gneiss.Message;
+with Componolit.Gneiss.Message.Reader;
+with Componolit.Gneiss.Strings;
+
+package body Component with
+   SPARK_Mode
+is
+
+   type Unsigned_Char is mod 2 ** 8;
+   type C_String is array (Positive range <>) of Unsigned_Char;
+
+   procedure Event;
+
+   package Message_Client is new Gns.Message.Reader (Unsigned_Char, Positive, C_String, 4096, Event);
+
+   type Client_Registry is array (Integer range <>) of Gns.Message.Reader_Session;
+
+   Client  : Client_Registry (1 .. 3);
+   Message : Message_Client.Message_Buffer;
+   M_Str   : String (1 .. 4097);
```

Comment:
4097 elements?

Reply:
`M_Str` is passed to a C function calling `printf`. The largest possible message received from the message interface is 4096 byte so a null terminator needs to be added.